test/controllers/simulations_controller_test.rb

```diff
@@ -28,17 +28,269 @@ def teardown
              simulation_description: 'Just testing',
              simulation_binaries: binaries,
              executor: '',
-             simulation_input: {
+             simulation_input: [{
                  entities: [
                      {
                          parameters: []
                      }
                  ]
-             },
+             }],
              format: :json
          }
 
     assert_response :success, response.body
   end
 
+
+  test 'Not successful registration of simulation should' do
+    binaries = Rack::Test::UploadedFile.new(
+        Rails.root.join('test', 'fixtures', 'files', 'simulation_binaries.zip'),
+        'application/zip'
+    )
+
+    post :create,
+         {
+             simulation_name: 'simulation_one',
+             simulation_description: 'Just testing',
+             simulation_binaries: binaries,
+             executor: '',
+             simulation_input: {
+                 entities: 1
+             },
+             format: :json
+         }
+
+    assert_response 500, response.body
```

Comment:
Jeśli w controllerze będzie rzucane validation error, to wszędzie będzie kod 412.